### PR TITLE
VRSb instruction memory reference needs separate index register

### PR DESCRIPTION
**Pull request by NigelYiboYu**
This pull request contains two commits:

1. The VRSb instruction format does not support index fields.
Fix the generateVRSbInstruction() API so it separates the memory
references index register. 

2.  Add an assert check to make sure the right node is used to generate memory reference. 
The generateS390MemoryReference(TR::Node*, TR::CodeGenerator*,bool)
API uses a constructor that handles load and store nodes with symbol
references. If the node has a no symbol reference, the compiler can crash in binary encoding.

Signed-off-by: Nigel Yu <yunigel@ca.ibm.com>

### CODE REVIEW

**Comment by NigelYiboYu** [comment]
This change is being tested. Marking it WIP.

File: compiler/z/codegen/OMRMemoryReference.cpp [hunk 1]
```diff
@@ -3696,6 +3696,10 @@ generateS390MemoryReference(TR::Register * br, int32_t disp, TR::CodeGenerator *
 TR::MemoryReference *
 generateS390MemoryReference(TR::Node * node, TR::CodeGenerator * cg, bool canUseRX)
    {
+   // The TR::MemoryReference(TR::Node*, TR::CodeGenerator*, bool); constructor (below) is made for store or load nodes
+   // that have symbol references. If the node has no sym ref, this memory reference's symbolRef is null, whichs can lead
```
Comment (fjeremic) [inline compiler/z/codegen/OMRMemoryReference.cpp]: Typo on "whichs". Also can you describe what kind of problems can be encountered in the binary encoding phase? i.e. who will need this symbol reference.

Comment (NigelYiboYu) [reply compiler/z/codegen/OMRMemoryReference.cpp]: Fixed.

File: compiler/z/codegen/OMRMemoryReference.cpp [hunk 2]
```diff
@@ -3696,6 +3696,12 @@ generateS390MemoryReference(TR::Register * br, int32_t disp, TR::CodeGenerator *
 TR::MemoryReference *
 generateS390MemoryReference(TR::Node * node, TR::CodeGenerator * cg, bool canUseRX)
    {
+   // The TR::MemoryReference(TR::Node*, TR::CodeGenerator*, bool); constructor (below) is made for store or load nodes
+   // that have symbol references.
+   // A symbol reference is needed to adjust memory reference displacement (TR::MemoryReference::calcDisplacement).
+   // If the node has no sym ref, this memory reference's gets a NULL symbol reference, which can lead to crashes in displacement
+   // calculations.
```
Comment (joransiu) [inline compiler/z/codegen/OMRMemoryReference.cpp]: Based on the comment, would it make more sense to push the TR_ASSERT to the constructor instead?

**Comment by fjeremic** [comment]
@NigelYiboYu if this is ready for merging please remove the WIP and I'll do a final review.

**Comment by NigelYiboYu** [comment]
Tests passed. Ready to merge.

File: compiler/z/codegen/OMRMemoryReference.cpp (hunk 2 above)
Comment (NigelYiboYu) [reply compiler/z/codegen/OMRMemoryReference.cpp]: This has been done.

**Comment by fjeremic** [comment]
@genie-omr build zlinux, zos

**Pull request merged**